### PR TITLE
Better pregional

### PR DESCRIPTION
**Pull request by YoniKiriaty**
pregional was weird, this branch is better

### CODE REVIEW

File: src/main/java/edu/greenblitz/robotName/OI.java [hunk 1]
```diff
@@ -68,7 +82,24 @@ public SmartJoystick getFourthJoystick() {
...
+		));
+		secondJoystick.R1.whileTrue(new CollectNote());
+		secondJoystick.POV_DOWN.whileTrue(new ParallelRaceGroup(
+				new RunFunnelByPower(-0.2),new WaitCommand(0.5)));
+		secondJoystick.L1.whileTrue(new RunFlyWheelByVelocityConstant().andThen(new RunFunnelByPower(0.5)));
+		secondJoystick.POV_UP.onTrue(new RunFunnelByPower(0.5));
+		secondJoystick.X.whileTrue(new MovePivotToAngle(Rotation2d.fromDegrees(65)));
+		secondJoystick.A.whileTrue(new PanicMode());
+		secondJoystick.POV_LEFT.whileTrue(new ReverseRunIntake());
+		secondJoystick.Y.whileTrue(new RunFlyWheelByPower(-0.2));
```
Comment (bossyshrimp68) [inline src/main/java/edu/greenblitz/robotName/OI.java:102]: all of this needs to be romyButtons and schoriButtons

File: src/main/java/edu/greenblitz/robotName/Robot.java [hunk 2]
```diff
@@ -61,37 +62,38 @@ public static RobotType getRobotType() {
 
 	@Override
 	public void robotInit() {
-		Pathfinding.setPathfinder(new LocalADStar());
+//		Pathfinding.setPathfinder(new LocalADStar());
 		CommandScheduler.getInstance().enable();
-		initializeLogger();
-		initializeAutonomousBuilder();
+//		initializeLogger();
+//		initializeAutonomousBuilder();
```
Comment (bossyshrimp68) [inline src/main/java/edu/greenblitz/robotName/Robot.java]: ...give back

File: src/main/java/edu/greenblitz/robotName/Robot.java [hunk 3]
```diff
@@ -61,37 +62,38 @@ public static RobotType getRobotType() {
...
 
-		Elbow.init();
-		Wrist.init();
-		Roller.init();
+//		Elbow.init();
+//		Wrist.init();
+//		Roller.init();
 
-		Lifter.init();
+//		Lifter.init();
```
Comment (bossyshrimp68) [inline src/main/java/edu/greenblitz/robotName/Robot.java]: ...

File: src/main/java/edu/greenblitz/robotName/Robot.java [hunk 4]
```diff
@@ -152,6 +155,6 @@ private void initializeLogger() {
 
 	@Override
 	public void autonomousInit() {
-		AutonomousSelector.getInstance().getChosenValue().schedule();
+//		AutonomousSelector.getInstance().getChosenValue().schedule();
```
Comment (bossyshrimp68) [inline src/main/java/edu/greenblitz/robotName/Robot.java]: ..

File: src/main/java/edu/greenblitz/robotName/commands/PanicMode.java [hunk 5]
```diff
@@ -6,14 +6,17 @@
...
 
     public PanicMode() {
         super(
                 new MoveShooterToAngle(PivotConstants.PresetPositions.SAFE.ANGLE),
-                new MoveElbowAndWristToSafe().alongWith(new MovePivotToAngle(PivotConstants.PresetPositions.SAFE.ANGLE)),
-                () -> Elbow.getInstance().isInShooterCollisionRange()
+                new InstantCommand(),
+//                new MoveElbowAndWristToSafe().alongWith(new MovePivotToAngle(PivotConstants.PresetPositions.SAFE.ANGLE)),
+                () -> true
+//               () -> Elbow.getInstance().isInShooterCollisionRange()() -> false
```
Comment (bossyshrimp68) [inline src/main/java/edu/greenblitz/robotName/commands/PanicMode.java]: ...

File: src/main/java/edu/greenblitz/robotName/commands/shooter/ShootFromInFunnel.java [hunk 6]
```diff
@@ -7,7 +7,7 @@ public class ShootFromInFunnel extends SequentialCommandGroup {
 
     public ShootFromInFunnel() {
         super(
-                new RunFlyWheelByVelocityConstant(),
+//                new RunFlyWheelByVelocityConstant(),
```
Comment (bossyshrimp68) [inline src/main/java/edu/greenblitz/robotName/commands/shooter/ShootFromInFunnel.java]: ...

File: src/main/java/edu/greenblitz/robotName/commands/shooter/ShootToSpeakerFromClose.java [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+import edu.greenblitz.robotName.commands.shooter.pivot.MovePivotToAngle;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.wpilibj2.command.SequentialCommandGroup;
+
+public class ShootToSpeakerFromClose extends SequentialCommandGroup {
+
+    public ShootToSpeakerFromClose() {
+        super(
+                new RunFlyWheelByVelocityConstant().alongWith(new MovePivotToAngle(Rotation2d.fromDegrees(45))),
+                new ShootFromInFunnel());
```
Comment (bossyshrimp68) [inline src/main/java/edu/greenblitz/robotName/commands/shooter/ShootToSpeakerFromClose.java:13]:
```suggestion
                new ShootFromInFunnel()
         );
```

File: src/main/java/edu/greenblitz/robotName/commands/shooter/ShootToSpeakerFromClose.java [hunk 8]
```diff
@@ -0,0 +1,15 @@
...
+import edu.greenblitz.robotName.commands.shooter.flyWheel.RunFlyWheelByVelocityConstant;
+import edu.greenblitz.robotName.commands.shooter.pivot.MovePivotToAngle;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.wpilibj2.command.SequentialCommandGroup;
+
+public class ShootToSpeakerFromClose extends SequentialCommandGroup {
+
+    public ShootToSpeakerFromClose() {
+        super(
+                new RunFlyWheelByVelocityConstant().alongWith(new MovePivotToAngle(Rotation2d.fromDegrees(45))),
```
Comment (bossyshrimp68) [inline src/main/java/edu/greenblitz/robotName/commands/shooter/ShootToSpeakerFromClose.java:12]: magic number

File: src/main/java/edu/greenblitz/robotName/subsystems/limelight/MultiLimelight.java [hunk 9]
```diff
@@ -36,7 +36,7 @@ public static void init() {
 	public List<Optional<Pair<Pose2d, Double>>> getAll2DEstimates() {
 		ArrayList<Optional<Pair<Pose2d, Double>>> estimates = new ArrayList<>();
 		for (Limelight limelight : limelights) {
-			if (limelight.hasTarget() && Math.abs(VisionConstants.APRIL_TAG_HEIGHT_METERS - limelight.getTagHeight()) < VisionConstants.APRIL_TAG_HEIGHT_TOLERANCE_METERS) {
+			if (limelight.hasTarget()){//&& Math.abs(VisionConstants.APRIL_TAG_HEIGHT_METERS - limelight.getTagHeight()) < VisionConstants.APRIL_TAG_HEIGHT_TOLERANCE_METERS) {
```
Comment (bossyshrimp68) [inline src/main/java/edu/greenblitz/robotName/subsystems/limelight/MultiLimelight.java:39]: .........................................

File: src/main/java/edu/greenblitz/robotName/Robot.java [hunk 10]
```diff
@@ -69,6 +70,7 @@ public void robotInit() {
 		SwerveChassis.getInstance().resetAngularEncodersByAbsoluteEncoder();
 		Dashboard.getInstance();
 		OI.init();
+		Pivot.getInstance().resetAngle(Rotation2d.fromDegrees(18));
```
Comment (bossyshrimp68) [inline src/main/java/edu/greenblitz/robotName/Robot.java:73]: magic number

File: src/main/java/edu/greenblitz/robotName/subsystems/swerve/modules/mk4iSwerveModule/MK4iSwerveConstants.java [hunk 11]
```diff
@@ -73,10 +73,10 @@ public class MK4iSwerveConstants {
 	
 	static {
 		Slot0Configs PID_CONFIGS = new Slot0Configs();
-		PID_CONFIGS.kS = ks;
-		PID_CONFIGS.kA = ka;
-		PID_CONFIGS.kV = kv;
-		PID_CONFIGS.kP = 1;
+		PID_CONFIGS.kS = 0.16411;
+		PID_CONFIGS.kV = 2.6824;
```
Comment (bossyshrimp68) [inline src/main/java/edu/greenblitz/robotName/subsystems/swerve/modules/mk4iSwerveModule/MK4iSwerveConstants.java:77]: ka and then kv

**Review by bossyshrimp68 — CHANGES_REQUESTED** [review]

File: src/main/java/edu/greenblitz/robotName/VisionConstants.java [hunk 12]
```diff
@@ -12,13 +12,13 @@ public class VisionConstants {
...
 	
-	public static double STANDARD_DEVIATION_ODOMETRY = 1;
+	public static double STANDARD_DEVIATION_ODOMETRY = 0.1;
 	
 	public static int LIMELIGHT_ENTRY_ARRAY_LENGTH = 7;
 	
 	public static double STANDARD_DEVIATION_ODOMETRY_ANGLE = 0.01;
 	
-	public static double STANDARD_DEVIATION_VISION2D = 0.5;
+	public static double STANDARD_DEVIATION_VISION2D = 0;
```
Comment (yonnbro) [inline src/main/java/edu/greenblitz/robotName/VisionConstants.java:21]: Are you sure?

**Review by yonnbro — APPROVED** [review]

File: src/main/java/edu/greenblitz/robotName/OI.java [hunk 13]
```diff
@@ -68,7 +82,31 @@ public SmartJoystick getFourthJoystick() {
...
```
Comment (bossyshrimp68) [inline src/main/java/edu/greenblitz/robotName/OI.java:109]: soooooo many magic numbers

File: src/main/java/edu/greenblitz/robotName/VisionConstants.java (hunk 12 above)
Comment (YoniKiriaty) [reply src/main/java/edu/greenblitz/robotName/VisionConstants.java:21]: yes

File: src/main/java/edu/greenblitz/robotName/subsystems/limelight/MultiLimelight.java [hunk 14]
```diff
@@ -36,7 +36,7 @@ public static void init() {
 	public List<Optional<Pair<Pose2d, Double>>> getAll2DEstimates() {
 		ArrayList<Optional<Pair<Pose2d, Double>>> estimates = new ArrayList<>();
 		for (Limelight limelight : limelights) {
-			if (limelight.hasTarget() && Math.abs(VisionConstants.APRIL_TAG_HEIGHT_METERS - limelight.getTagHeight()) < VisionConstants.APRIL_TAG_HEIGHT_TOLERANCE_METERS) {
+			if (limelight.hasTarget()){
```
Comment (bossyshrimp68) [inline src/main/java/edu/greenblitz/robotName/subsystems/limelight/MultiLimelight.java:39]:
```suggestion
			if (limelight.hasTarget()) {
```

**Review by bossyshrimp68 — CHANGES_REQUESTED** [review]

**Review by yonnbro — APPROVED** [review]

File: src/main/java/edu/greenblitz/robotName/OI.java [hunk 15]
```diff
@@ -68,7 +82,25 @@ public SmartJoystick getFourthJoystick() {
...
+
+	public void romyButtons() {
+		mainJoystick.R1.whileTrue(new CollectNote());
+		mainJoystick.L1.whileTrue(new MoveRobotToShootingPosition(ShootingPositionConstants.OPTIMAL_SHOOTING_ZONE));
+		mainJoystick.A.onTrue(new InstantCommand(() -> SwerveChassis.getInstance().resetPoseByVision()));
+	}
 
+	public void schoriButtons() {
+		secondJoystick.BACK.whileTrue(new InstantCommand(() -> secondJoystick.rumble(true,1)));
+		secondJoystick.START.whileTrue(new InstantCommand(() -> secondJoystick.rumble(true,0)));
```
Comment (bossyshrimp68) [inline src/main/java/edu/greenblitz/robotName/OI.java:97]:
```suggestion
		secondJoystick.BACK.whileTrue(new InstantCommand(() -> secondJoystick.rumble(true, 1)));
		secondJoystick.START.whileTrue(new InstantCommand(() -> secondJoystick.rumble(true, 0)));
```

File: src/main/java/edu/greenblitz/robotName/subsystems/limelight/MultiLimelight.java [hunk 16]
```diff
@@ -36,7 +36,7 @@ public static void init() {
 	public List<Optional<Pair<Pose2d, Double>>> getAll2DEstimates() {
 		ArrayList<Optional<Pair<Pose2d, Double>>> estimates = new ArrayList<>();
 		for (Limelight limelight : limelights) {
-			if (limelight.hasTarget() && Math.abs(VisionConstants.APRIL_TAG_HEIGHT_METERS - limelight.getTagHeight()) < VisionConstants.APRIL_TAG_HEIGHT_TOLERANCE_METERS) {
```
Comment (bossyshrimp68) [inline src/main/java/edu/greenblitz/robotName/subsystems/limelight/MultiLimelight.java:39]: give it back

**Review by bossyshrimp68 — CHANGES_REQUESTED** [review]

**Review by bossyshrimp68 — APPROVED** [review]

**Review by yonnbro — APPROVED** [review]

**Pull request merged**